UE4SS/proxy_generator/main.cpp

```diff
@@ -176,13 +181,73 @@ int _tmain(int argc, TCHAR* argv[])
+    cpp_file << endl;
+
+    cpp_file << "    // Check for override.txt" << endl;
+    cpp_file << "    const fs::path overrideFilePath = currentPath / \"override.txt\";" << endl;
+    cpp_file << "    if (fs::exists(overrideFilePath))" << endl;
+    cpp_file << "    {" << endl;
+    cpp_file << "        std::ifstream overrideFile(overrideFilePath);" << endl;
+    cpp_file << "        std::string overridePath;" << endl;
+    cpp_file << "        if (std::getline(overrideFile, overridePath))" << endl;
+    cpp_file << "        {" << endl;
+    cpp_file << "            fs::path ue4ssOverridePath = overridePath;" << endl;
+    cpp_file << "            if (!is_absolute_path(overridePath))" << endl;
+    cpp_file << "            {" << endl;
+    cpp_file << "                ue4ssOverridePath = currentPath / overridePath;" << endl;
+    cpp_file << "            }" << endl;
+    cpp_file << endl;
+    cpp_file << "            ue4ssOverridePath = ue4ssOverridePath / \"UE4SS.dll\";" << endl;
+    cpp_file << endl;
+    cpp_file << "            // Attempt to load UE4SS.dll from the override path" << endl;
+    cpp_file << "            hModule = LoadLibrary(ue4ssOverridePath.c_str());" << endl;
+    cpp_file << "            if (hModule)" << endl;
+    cpp_file << "            {" << endl;
+    cpp_file << "                return hModule;" << endl;
+    cpp_file << "            }" << endl;
+    cpp_file << "        }" << endl;
+    cpp_file << "    }" << endl;
+    cpp_file << endl;
+
+    cpp_file << "    // Attempt to load UE4SS.dll from ue4ss directory" << endl;
+    cpp_file << "    hModule = LoadLibrary(ue4ssPath.c_str());" << endl;
+    cpp_file << "    if (!hModule)" << endl;
+    cpp_file << "    {" << endl;
+    cpp_file << "        // If loading from ue4ss directory fails, load from the current directory" << endl;
+    cpp_file << "        hModule = LoadLibrary(STR(\"UE4SS.dll\"));" << endl;
+    cpp_file << "    }" << endl;
+    cpp_file << endl;
+    cpp_file << "    return hModule;" << endl;
+    cpp_file << "}" << endl;
+    cpp_file << endl;
+
     cpp_file << "BOOL WINAPI DllMain(HMODULE hInstDll, DWORD fdwReason, LPVOID lpvReserved)" << endl;
     cpp_file << "{" << endl;
     cpp_file << "    if (fdwReason == DLL_PROCESS_ATTACH)" << endl;
     cpp_file << "    {" << endl;
     cpp_file << "        load_original_dll();" << endl;
-    cpp_file << "        setup_functions();" << endl;
-    cpp_file << "        LoadLibrary(STR(\"UE4SS.dll\"));" << endl;
+    cpp_file << "        HMODULE hUE4SSDll = load_ue4ss_dll(hInstDll);" << endl;
+    cpp_file << "        if (hUE4SSDll)" << endl;
+    cpp_file << "        {" << endl;
+    cpp_file << "            setup_functions();" << endl;
+    cpp_file << "        }" << endl;
+    cpp_file << "        else" << endl;
+    cpp_file << "        {" << endl;
+    cpp_file << "            ExitProcess(0);" << endl;
```

Comment:
Should we put a MessageBox here maybe to explain what happened to the user ?

Reply:
I put in two; one for when proxy fails to load for whatever reason, and one for when UE4SS.dll loads. Not sure about the error message, currently I just have "Failed to load UE4SS.dll" but don't know if it should be any more informative. Like I need to write up docs on the override file and such, so maybe the message can point to read that to make sure it's installed in valid locations.

Reply:
Yeah that might be good.

Reply:
We should put some kind of prefix or title or something on these message boxes that tells users that this message is from UE4SS.